packages/schema/src/config/build.ts

```diff
@@ -59,6 +59,27 @@ export default defineUntypedSchema({
       $resolve: val => [].concat(val).filter(Boolean)
     },
 
+    /**
+     * Tree shake composables from the server or client builds.
+     *
+     * @example
+     * ```js
+     * treeShake: { server: ['useClientOnlyComposable'] }
+     * ```
+     */
+    treeShake: {
+      server: {
+        $resolve: async (val, get) => (await get('dev') ? [] : [
+          'onBeforeMount', 'onMounted', 'onBeforeUpdate', 'onRenderTracked', 'onRenderTriggered', 'onActivated', 'onDeactivated', 'onBeforeUnmount'
+        ]).concat(val).filter(Boolean)
+      },
+      client: {
+        $resolve: async (val, get) => (await get('dev') ? [] : [
+          'onServerPrefetch', 'onRenderTracked', 'onRenderTriggered'
+        ]).concat(val).filter(Boolean)
+      }
+    },
```

Comment:
I wonder if it's possible to hard code the unhead composables we need without exposing this in the build config, at least as-of yet....

Reply:
This is definitely possible for the Unhead composables as they are already core to nuxt, the issue is that I need this for some other modules:
- nuxt-schema-org - `useSchemaOrg`
- nuxt-og-image - `defineOgImageStatic`, `defineOgImageDynamic`, `defineOgImageScreenshot`
- nuxt-simple-robots - `defineRobotMeta`
- ...probably some others

If you have any ideas on how to solve this without exposing config then happy to explore it

Reply:
We thought a new top level `optimization` option would be good. (We can include it in the same `build.ts` schema file.) It can include both this config (`optimization.treeshake.composables.client` & `.server`) and also future config like https://github.com/nuxt/nuxt/issues/12369.